src/extension/iSamples_pagination.js

```diff
@@ -0,0 +1,77 @@
+import PropTypes from 'prop-types';
+import React from "react";
+import cx from "classnames";
+
+
+class Pagination extends React.Component {
+
+  onPageChange(page, pageAmt) {
+    if (page >= pageAmt || page < 0) {
+      return;
+    }
+    this.props.onChange(page);
+  }
+
+  renderPage(page, currentPage, key) {
+    return (
+      <li className={cx({ "active": page === currentPage })} key={key}>
+        <a href="/#" onClick={this.onPageChange.bind(this, page)}>{page + 1}</a>
+      </li>
+    );
+  }
+
+
+  render() {
+    const { bootstrapCss, query, results } = this.props;
+    const { start, rows } = query;
+    const { numFound } = results;
+    const pageAmt = Math.ceil(numFound / rows);
+    const currentPage = start / rows;
+
+    let rangeStart = currentPage - 2 < 0 ? 0 : currentPage - 2;
+    let rangeEnd = rangeStart + 5 > pageAmt ? pageAmt : rangeStart + 5;
+
+    if (rangeEnd - rangeStart < 5 && rangeStart > 0) {
+      rangeStart = rangeEnd - 5;
+      if (rangeStart < 0) {
+        rangeStart = 0;
+      }
+    }
+
+    let pages = [];
+    for (let page = rangeStart; page < rangeEnd; page++) {
+      if (pages.indexOf(page) < 0) {
+        pages.push(page);
+      }
+    }
+
+    return (
+      <div className={cx({ "panel-body": bootstrapCss, "text-center": bootstrapCss })}>
+        <ul className={cx("pagination", "pagDisplay", { "pagination-sm": bootstrapCss })}>
```

Comment:
Add a class to the page number elements so we could hide or display it for the different views.